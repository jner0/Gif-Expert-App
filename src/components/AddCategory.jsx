import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ( { onNewCategory } ) => {

  const [inputValue, setInputValue] = useState('');

  const onInputChange = (/*event*/ { target }) => {
    setInputValue( target.value );
  }

  const onSubmit = ( event ) => {
    console.log('Hola Mundo desde onSubmit');
    event.preventDefault();
    if(inputValue.trim().length <= 1) return;

    //setCategories( categories => [ inputValue, ...categories  ]);
    onNewCategory( inputValue.trim() );
    setInputValue('');

  }

  return (

    <form onSubmit={ (event ) => onSubmit(event)} aria-label="form" >
        <input 
            type="text"
            placeholder="Buscar Gif"
            value={ inputValue }
            onChange={ onInputChange }
        
        />
    </form>
    
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
