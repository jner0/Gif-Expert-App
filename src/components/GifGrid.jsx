import { useEffect, useState } from "react";

import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
//import { getGifs } from "../helpers/getGifs";

export const GifGrid = ({category}) => {

    //lo Haremos creando un custom Hook
    const { images, isLoading } = useFetchGifs( category );
   

    // const [images, setImages] = useState([]);

    // const getImages = async() => {
    //     const newImages = await getGifs( category );
    //     setImages( newImages );
    // }

    // useEffect( () => {

    //     getImages();

    // }, []);


    return (
        <>
            <h3>{ category }</h3>
            {
                isLoading
                ? ( <h2>Cargando...</h2> )
                : null
            }

            <div className="card-grid">
                {
                    images.map( (image ) => (
                        <GifItem 
                            key={ image.id }
                            { ...image } //otra forma de hacerlo, todas las propiedades que vengan en el imagen las estamos esparciendo
                            // title={ image.title }
                            // url={ image.url }
                        />
                    ))
                }
            </div>
            
        </>
    )
}
