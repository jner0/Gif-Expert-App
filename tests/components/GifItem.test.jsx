import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en GifItem', () => { 

    const title = "Tommy";
    const url = "https://peaky-blinders.com/tommy.jpg";

    test('Debe de hacer match con el snapshot', () => { 
    
        const { container } =  render( <GifItem title={title} url={url}/> );
        expect( container ).toMatchSnapshot();

     });

    test('Debe de mostrar la imagen con el url y alt indicado', () => { 

        render( <GifItem title={title} url={url}/> );
        //screen.debug();
        // expect( screen.getByRole('img').src).toBe(url);
        const { src, alt } = screen.getByRole('img');
        expect( src ). toBe(url);
        expect( alt ).toBe(alt);


     });

     test('Debe de mostrar el titulo en el componente', () => { 

        render( <GifItem title={title} url={url}/> );
        expect( screen.getByText( title )).toBeTruthy();

     })

 });