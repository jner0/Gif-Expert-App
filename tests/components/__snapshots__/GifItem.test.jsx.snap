// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Tommy"
      src="https://peaky-blinders.com/tommy.jpg"
    />
    <p>
      Tommy
    </p>
  </div>
</div>
`;
